From 82d8584b889db5166bf7ad7863e079d38c9c6e7a Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Sat, 2 Sep 2023 15:28:31 +0200
Subject: [PATCH 5/6] Revert "ASoC: Intel: soc-acpi: add tables for Dell SKU
 0B34"

This reverts commit b62a1a839b48f55046727089c3ba7a8ebbf97f0e.

It broke stereo sound on the Dell XPS 13 Plus 9320 (SKU 0AF3),
see https://github.com/thesofproject/linux/issues/4559.

Considering that this apparently wasn't enough to get sound working on
SKU 0B34 (https://github.com/thesofproject/linux/issues/4399) and that
Dell officially supports Linux on 0AF3 but not 0B34, I think this is
safe to revert.

Fixes: https://github.com/thesofproject/linux/issues/4559
Signed-off-by: Jan Alexander Steffens (heftig) <heftig@archlinux.org>
---
 .../intel/common/soc-acpi-intel-adl-match.c   | 29 -------------------
 1 file changed, 29 deletions(-)

diff --git a/sound/soc/intel/common/soc-acpi-intel-adl-match.c b/sound/soc/intel/common/soc-acpi-intel-adl-match.c
index bcd66e0094b4..83414bfa8d6c 100644
--- a/sound/soc/intel/common/soc-acpi-intel-adl-match.c
+++ b/sound/soc/intel/common/soc-acpi-intel-adl-match.c
@@ -133,15 +133,6 @@ static const struct snd_soc_acpi_adr_device rt1316_1_group2_adr[] = {
 	}
 };
 
-static const struct snd_soc_acpi_adr_device rt1316_1_single_adr[] = {
-	{
-		.adr = 0x000130025D131601ull,
-		.num_endpoints = 1,
-		.endpoints = &single_endpoint,
-		.name_prefix = "rt1316-1"
-	}
-};
-
 static const struct snd_soc_acpi_adr_device rt1316_2_single_adr[] = {
 	{
 		.adr = 0x000230025D131601ull,
@@ -321,20 +312,6 @@ static const struct snd_soc_acpi_link_adr adl_sdw_rt1316_link12_rt714_link0[] =
 	{}
 };
 
-static const struct snd_soc_acpi_link_adr adl_sdw_rt1316_link1_rt714_link0[] = {
-	{
-		.mask = BIT(1),
-		.num_adr = ARRAY_SIZE(rt1316_1_single_adr),
-		.adr_d = rt1316_1_single_adr,
-	},
-	{
-		.mask = BIT(0),
-		.num_adr = ARRAY_SIZE(rt714_0_adr),
-		.adr_d = rt714_0_adr,
-	},
-	{}
-};
-
 static const struct snd_soc_acpi_link_adr adl_sdw_rt1316_link2_rt714_link3[] = {
 	{
 		.mask = BIT(2),
@@ -648,12 +625,6 @@ struct snd_soc_acpi_mach snd_soc_acpi_intel_adl_sdw_machines[] = {
 		.drv_name = "sof_sdw",
 		.sof_tplg_filename = "sof-adl-rt1316-l2-mono-rt714-l3.tplg",
 	},
-	{
-		.link_mask = 0x3, /* rt1316 on link1 & rt714 on link0 */
-		.links = adl_sdw_rt1316_link1_rt714_link0,
-		.drv_name = "sof_sdw",
-		.sof_tplg_filename = "sof-adl-rt1316-l1-mono-rt714-l0.tplg",
-	},
 	{
 		.link_mask = 0x7, /* rt714 on link0 & two rt1316s on link1 and link2 */
 		.links = adl_sdw_rt1316_link12_rt714_link0,
-- 
2.42.0

